### ISSUE TITLE
Specify different cache locations via conanfile.py?

### ISSUE THREAD

**Issue by alreadyhaveanaccount4** [conan-io/conan#16041]
### What is your question?

_version: 1.63.0
os: linux
conanfile: conanfile.py_

Hi there!

we have the following use-case:
in the CI, after the build, we upload a latest version of the cache of each profile to a remote storage. We then want to use these caches (e.g. cache-profile-1, cache-profile-2, etc.) locally. We have a script to download them and put them to a specific directory, e.g.:
`profile_1 -> ~/.my-conan-cache/cache-profile-1/.conan/`
`profile_2 -> ~/.my-conan-cache/cache-profile-2/.conan/`
`etc.`

Is there a way via conanfile.py to set the directory for the cache? 

In short:
when running e.g. `conan install . -pr:h=profiles/profile-1`, can i somehow automatically tell conan where to look for the cache? Something like this
```
def set_cache_dir(self):
      self.download_cache = "/home/user/.my-conan-cache/cache-{self.current_profile}/.conan/"
```

I know the `CONAN_USER_HOME` variable, but it would be a bit of a hickup to automate this.
I am fairly new to conan, so I am thankful for any advice.

Best regards

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @alreadyhaveanaccount4 

Thanks for your question.

In Conan 2 you have the ``.conanrc`` file that can automatically set the ``CONAN_HOME`` home folder. So it would be enough to put each conanfile inside a folder with a ``.conanrc`` file besides it. Please have a look to https://docs.conan.io/2/reference/config_files/conanrc.html#conanrc and let us know if this helps.

**Comment by alreadyhaveanaccount4** [comment]
Hi @memsharded!

thanks for your quick response!

We use conan 1.63.0, so the .conanrc file approach wouldn't work for us right now, is there any alternative for 1.6x versions?

Regarding your hint for conan2:
We don't have multiple conanfiles/projects, but multiple profiles for one project with a single conanfile. We want to have seperate caches for each profile. I am not sure if the .conanrc would help us (when we upgrade to conan2). So when running `conan install . -pr:h=profiles/profile-1`, it should take the root `conanfile.py`, `profile-1` and the cache under `/home/user/.my-conan-cache/cache-profile-1`. Similar for profile-2. With the .conanrc, we would have to have profile-specific folders, each containing the .rc file + profile and execute conan commands from there. Correct me if I am wrong?

```
project/
 - conanfile.py
 - profiles/
      - profile-1
      - profile-2
 - requirements.txt
```

**Comment by memsharded** [comment]
> We use conan 1.63.0, so the .conanrc file approach wouldn't work for us right now, is there any alternative for 1.6x versions?

I am afraid not, you might need to set CONAN_USER_HOME per job/terminal for it


> We don't have multiple conanfiles/projects, but multiple profiles for one project with a single conanfile. We want to have seperate caches for each profile.

Uhm, but why would  you want to do that? One of the advantages of the cache design is that this isn't necessary. A different story is concurrency, which the cache is not at the moment, so in concurrent environments like CI, it is necessary to define a new home per job, but that can be easily done in the CI itself.

It is not very clear what problem you are trying to solve with that design, one cache per profile, if you could please elaborate.

**Comment by memsharded** [comment]
Any further feedback here? If not, maybe we can close this question as responded?

**Comment by memsharded** [comment]
I am closing this ticket as responded, if you have any further question don't hesitate to open new tickets, thanks!
